src/cpp/rtps/DataSharing/DataSharingListener.cpp

```diff
@@ -217,11 +218,21 @@ bool DataSharingListener::add_datasharing_writer(
 
     std::shared_ptr<ReaderPool> pool =
             std::static_pointer_cast<ReaderPool>(DataSharingPayloadPool::get_reader_pool(is_volatile));
-    pool->init_shared_memory(writer_guid, datasharing_pools_directory_);
-    writer_pools_.emplace_back(pool, pool->last_liveliness_sequence());
-    writer_pools_changed_.store(true);
+    if (pool->init_shared_memory(writer_guid, datasharing_pools_directory_))
+    {
+        if (0 == reader_history_max_samples ||
+                reader_history_max_samples > static_cast<int32_t>(pool->history_size() - 1))
```

Comment:
Negative values also mean infinite.
`A > (B - 1)` ==> `A >= B`

```suggestion
        if (0 <= reader_history_max_samples ||
                reader_history_max_samples >= static_cast<int32_t>(pool->history_size()))
```

Reply:
```suggestion
        if (0 >= reader_history_max_samples ||
                reader_history_max_samples >= static_cast<int32_t>(pool->history_size()))
```

Reply:
Should be in this way to catch infinitive values?

Reply:
The first condition is for the case where `reader_history_max_samples` is infinite. I'm assuming that a negative value also means infinite.

On the second condition, I think that `pool->history_size()` will always return a strictly positive value, as it is the way the pool is designed to be.

Reply:
Now I see what you mean! It is `0 >= reader_history_max_samples`, not `0 <= reader_history_max_samples` !